### DISCUSSION TITLE
mysql_script module

### DISCUSSION THREAD

**Discussion by Andersson007** [ansible-collections/ansible.mysql#285] [Ideas]
What do you folks think of creating the `mysql_script` module?
This feature is popular among `community.postgresql` collection users but it's part of `postgresql_query` module and we are thinking of moving the functionality to the dedicated `postgresql_script` module. See https://github.com/ansible-collections/community.postgresql/issues/180#issuecomment-1033568892 and nearby.
Also I've created a PR https://github.com/ansible-collections/community.postgresql/pull/188

So there are several questions
1. How about having a similar module here?
2. (provided that the answer on the above question is yes) How should it be called? My option is `mysql_script`
3. How about it's interface (i.e. arguments, their possible values, and return values functionality)?

It can be, for example
```
- name: Run statements from the file
  mysql_script / mysql_file_queries:
    path / src: /path/to/file
    (positional_args:) # maybe
    (named_args:) # maybe
```
Return values can be similar to `mysql_query` but if the file contains several queries, the module will return data relevant only for the last query

@bmalynovytch @Jorge-Rodriguez @rsicart 

(I'm not in favor of not idempotent stuff but it seems useful for users)

**Comment by rsicart** [comment]
Hi there!

I understand that having a module for executing a script can be interesting (perhaps useful to reduce integration tests lines of code too?), but are users of mysql collection asking for this feature?

If I had to choose a name, I'd go for mysql_script name too.

**Reply by Andersson007** [reply]
> but are users of mysql collection asking for this feature?

No, they aren't:)
This is a really good question. My suggestion is based on the popularity of the similar feature in `community.postgresql`.
I'm not in favor of introducing this, nor against, just interesting whether it could be useful:)
Users also didn't request `mysql_query` or `mysql_info` modules in this collection but they have got pretty popular since we introduced them a couple of years ago:)
Based on our experience with c.postgres, we can expect that users will like the feature BUT, yeah, this would be a piece of code someone needs to maintain.

**Comment by Andersson007** [comment]
Maybe other ideas of new modules/plugins?
Yes, users don't request any new stuff (the last was `mysql_role` which is implemented) but maybe there's some (new) features that are not covered by the collection yet.

**Comment by supriyoss** [comment]
@Andersson007 @rsicart just when I was searching on Google for the feature, this discussion thread came up. I'm in urgent need of the module. Having this module, it'll help me a lot. I need to automate some deployments for my client and during deployments, they execute a series of sql scripts. If we had the mysql_scripts module life would have been a lot easier for me!!! Thanks.

**Reply by Andersson007** [reply]
@supriyoss hello, thanks for the feedback! Could you please take a look at the [similar module](https://docs.ansible.com/ansible/latest/collections/community/postgresql/postgresql_script_module.html) in postgresql collection and say if the interface is OK?

@rsicart @laurent-indermuehle @betanummeric @bmalynovytch @Jorge-Rodriguez (i hope i didn't forget anyone) what do you think? Also if the majority of us is happy, would anyone like to write the module? I guess `postgresql_script` can be used as a basis.

**Reply by laurent-indermuehle** [reply]
Maybe I'm wrong, but can't we already do that with mysql_db?

```yaml
- name: Import SQL script
  community.mysql.mysql_db:
    name: all
    state: import
    target: "script.sql"
```

**Reply by Andersson007** [reply]
@supriyoss what do you think on @laurent-indermuehle 's suggestion?
